tools/app_interface_reporter.py

```diff
@@ -387,10 +403,30 @@ def get_build_history(jenkins_map, jobs, timestamp_limit):
         jenkins = jenkins_map[instance]
         for job in jobs:
             logging.info(f"getting build history for {job['name']}")
-            build_history = \
-                jenkins.get_build_history(job['name'], timestamp_limit)
-            repo_url = get_repo_url(job)
-            history[repo_url] = build_history
+            # continue if no repo_url
+            if 'properties' not in job:
+                continue
+            if 'github' not in job['properties'][0]:
+                continue
+
+            if 'build_branch' in job:
+                job_env = job['build_branch']
+            elif 'id' in job:
+                job_env = job['id']
+            else:
+                continue
```

Comment:
Could you give some examples of these 3 kinds of jobs?

Reply:
`ci-int` and `ci-ext` jobs use `id` 
Example
```
'name': 'insights-platform-pes-cloud-gl-build-master',
'properties': [{'github': {'url': 'https://gitlab.cee.redhat.com/insights-platform/pes-cloud.git'}},
  {'gitlab': {'connection': 'gitlab cee'}}],
'id': 'gl-build-master'
```

while `ci-centos` use `build_branch`
Example:
```
'name':  'devtools-fabric8-notification-build-master-coverage',
'properties': [{'github': {'url': 'https://github.com/fabric8-services/fabric8-notification/'}}],
'build_branch': 'master'
```

Currently we do not have the case where neither `id` nor `build_branch` appears.

Reply:
Could you try with job['branch']?

Reply:
Perfect! All jobs has `branch` key but it seems that there are some small variations. For example, 
```
'instance' : ci-int
'name': dtsd-osio-scan-gl-build-master
'branch':  master
'id': gl-build-master-osio-scan
```
and 
```
'instance' : ci-ext
'name': app-sre-manifest-bouncer-gh-build-tag
'branch': refs/tags/*
'id': gh-build-tag
```
Do we care about those variations or shall we just go ahead and use job['branch']?

Reply:
id is used for matching with job-temple. It is not unique and it may not reflect the actual branch.

Reply:
updated!